p2p/host/basic/natmgr.go

```diff
@@ -21,6 +21,7 @@ import (
 // and tries to obtain port mappings for those.
 type NATManager interface {
 	GetMapping(ma.Multiaddr) ma.Multiaddr
+	HasNAT() bool
```

Comment:
I don't think we should call this `HasNAT` because this checks if we can communicate with a NAT device. If we have a NAT, but are unable to communicate with it (e.g. via upnp) then, this would return false.

Maybe call it `HasDiscoveredNAT` instead?